test-lit/unit/version_test.f90

```diff
@@ -1,4 +1,4 @@
 ! Check that lit configuration works by checking the compiler version
 
-! RUN: f18 -V 2>&1 | FileCheck  -check-prefix=VERSION %s
-! VERSION-DAG: f18 compiler
+! RUN: %f18 -V 2>&1 | FileCheck  -check-prefix=VERSION %s
```

Comment:
I don't like 'unit' as a test subdirectory name. I think it should be driver, as that is what this test is testing.

Reply:
This would still not catch lots of bad cases. All of the below version outputs would pass this test.

> echo "f18 compiler (under development) foo" | FileCheck --check-prefix=VERSION version_test.f90
> echo "foo f18 compiler (under development)" | FileCheck --check-prefix=VERSION version_test.f90
> echo "f18 compiler (under development)
> 
> foo" | FileCheck --check-prefix=VERSION version_test.f90

Reply:
Done!